mage/query-modules/python/temporal-graph-networks.md

```diff
@@ -0,0 +1,423 @@
+| node_label_property 	                | String 	     | `label` 	            |  name of label property on nodes from which we read features	|
+
+#### Usage:
+
+```cypher
+ CALL tgn.set_params({learning_type:'self_supervised', batch_size:200, num_of_layers:2, 
+                      layer_type:'graph_attn',memory_dimension:20, time_dimension:50, 
+                      num_edge_features:20, num_node_features:20, message_dimension:100, 
+                      num_neighbors:15, edge_message_function_type:'identity',
+                      message_aggregator_type:'last', memory_updater_type:'gru', num_attention_heads:1});
+```
+
+
+### `update(edges)`
+This function scrapes data from edge, including `edge_features` and `node_features` if they exist. It also fills up the batch, and once the batch is ready **TGN** will process
+batch, and be ready to accept new incoming edges.
+#### Input:
+
+* `edges: mgp.List[mgp.Edges]` ➡ list of edges that arrive in a stream to `Memgraph` database to preprocess. If a batch is full, `train` or `eval` starts, depending on the mode.
+
+#### Usage:
+
+There are a few options here:
+
+The most convenient one is to create a **[trigger](https://memgraph.com/docs/memgraph/reference-guide/triggers)** so that every
+time an edge is added to the graph, the trigger calls the procedure and makes an
+update.
+
+```cypher
+CREATE TRIGGER create_embeddings ON --> CREATE BEFORE COMMIT
+EXECUTE CALL tgn.update(createdEdges) RETURN 1;
+```
+
+The second option is to add all the edges and then call the algorithm with them:
+
+```cypher
+MATCH (n)-[e]->(m)
+WITH COLLECT(e) as edges
+CALL tgn.update(edges) RETURN 1;
+```
+
+
+### `get()`
+Get calculated embeddings for each vertex.
+#### Output:
+
+* `node: mgp.Vertex` ➡ vertex in Memgraph database
+* `embedding: mgp.List[float]` ➡ low-dimensional representation of node in form of graph embedding
+
+#### Usage:
+
+```cypher
+CALL tgn.get() YIELD * RETURN *;
+```
+
+
+### `set_eval()`
+Change **TGN* mode to `eval`.
+#### Usage:
+
+```cypher
+CALL tgn.set_eval() YIELD *;
+```
+
+
+### `get_results()`
+This method will return for `results` for every batch you did `train` or `eval` on, as well as `average_precision`, and `batch_process_time`. 
+Epoch count starts from 1.
+#### Output:
+
+* `epoch_num:mgp.Number` ➡ number of `train` or `eval` epochs
+* `batch_num:mgp.Number` ➡ number of batches per `train` or `eval` epoch
+* `batch_process_time:mgp.Number` ➡ time needed to process batch 
+* `average_precision:mgp.Number` ➡ mean average precision on the current batch
+* `batch_type:str` ➡ whether we performed `train` or `eval` on the batch
+
+
+#### Usage:
+
+```cypher
+CALL tgn.get_results() YIELD * RETURN *;
+```
+
+
+### `train_and_eval(num_epochs)`
+The purpose of this method is to do additional training rounds on `train` edges and `eval` on evaluation edges.
+#### Input:
+
+* `num_epochs: int` ➡ perform additional epoch training and evaluation **after** the stream is done
```

Comment:
I would move the "after the stream is done" part to the sentence just above **Input**.